src/shims/native_lib/mod.rs

```diff
@@ -439,7 +528,12 @@ trait EvalContextExtPriv<'tcx>: crate::MiriInterpCxExt<'tcx> {
         interp_ok(match layout.ty.kind() {
             // Scalar types have already been handled above.
             ty::Adt(adt_def, args) => self.adt_to_ffitype(layout.ty, *adt_def, args)?,
-            _ => throw_unsup_format!("unsupported argument type for native call: {}", layout.ty),
+            // Functions with no declared return type (i.e., the default return)
+            // have the output_type `Tuple([])`.
+            ty::Tuple(t_list) if (*t_list).deref().is_empty() => FfiType::void(),
```

Comment:
This should not be needed -- function pointers are scalar types.

Reply:
The function I happened to use for testing returned `void`. As this function is now also called to construct the return value types for the libffi closure type this was required to make my test case working. 

That written: You are correct that this is not needed for a minimal support of callback over fro, it just happens to help with my particular test case and was rather straightforward to add.

Reply:
> this function is now also called to construct the return value types for the libffi closure type 

Ah, that makes sense.

Given that I think we should never return from that closure (see my other comments), I think we should not need this either.

Reply:
Could handle this separately in another PR and make all tests here have an already implemented return type. But not too important

Reply:
I can do this, but in that case I likely also need to change existing tests to make it work with the implemented feature as they also use void as return/argument type for the callback.